subcommand/injector/command.go

```diff
@@ -206,11 +211,28 @@ func (c *Command) Run(args []string) int {
 	}
 
 	var handler http.Handler = mux
+	minTLSVersion, ok := tlsutil.TLSLookup[c.flagTLSMinVersion]
+	if !ok {
+		c.UI.Error(fmt.Sprintf("Failed to parse minimum TLS version %q", c.flagTLSMinVersion))
+		return 1
+	}
+	ciphers, err := tlsutil.ParseCiphers(c.flagTLSCipherSuites)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Failed to parse TLS cipher suites list %q: %s", c.flagTLSCipherSuites, err))
+		return 1
+	}
 	server := &http.Server{
-		Addr:      c.flagListen,
-		Handler:   handler,
-		TLSConfig: &tls.Config{GetCertificate: c.getCertificate},
-		ErrorLog:  logger.StandardLogger(&hclog.StandardLoggerOptions{ForceLevel: hclog.Error}),
+		Addr:    c.flagListen,
+		Handler: handler,
+		TLSConfig: &tls.Config{
+			GetCertificate:           c.getCertificate,
+			MinVersion:               minTLSVersion,
+			PreferServerCipherSuites: c.flagTLSPreferServerCipherSuites,
+		},
+		ErrorLog: logger.StandardLogger(&hclog.StandardLoggerOptions{ForceLevel: hclog.Error}),
+	}
+	if len(ciphers) > 0 {
+		server.TLSConfig.CipherSuites = ciphers
```

Comment:
nit: I feel like this change warrants factoring out the `TLSConfig{}` setup from `Server{}`